app/ui/permissions.py

```diff
@@ -53,5 +53,8 @@ def setup_permissions():
             None,
         )
     if not config.up_to_date():
-        with connection.Connection.get() as conn:
+        with connection.Connection.get() as conn, st.status(
+            "Post-upgrade tasks are running in the background..."
+        ) as status:
             conn.call(f"{db}.ADMIN.FINALIZE_SETUP")
+            status.update(label="Upgrade successful!")
```

Comment:
What has to still be run in finalize_setup? It would be good if a user didn't have to open up the UI every time the versions were updated. (can be part of a later PR)

Reply:
Aside: I should probably remove this and defer to a later change-- it's just "fixing" the blank page when finalize_setup is running post-upgrade.

From what I remember, the biggest piece of finalize_setup() are the table migration checks which take ~1-2minutes.

Reply:
that runs in the tasks too doesnt it? I don't remember, I just don't want to ahve a user open the UI after the first time

Reply:
I don't remember either, but I agree with your goal. I will take a look.

I think we will have to be extra careful of this going forward (ensuring any state in an unversioned schema is migrated automatically w/o the need to open the UI).